src/base/kaldi-utils.cc

```diff
@@ -45,6 +45,8 @@ std::string CharToString(const char &c) {
 void Sleep(float seconds) {
 #if defined(_MSC_VER) || defined(MINGW)
   ::Sleep(static_cast<int>(seconds * 1000.0));
+#elif defined(__CYGWIN__)
+  kaldi::Sleep(static_cast<int>(seconds * 1000.0));
```

Comment:
this makes me think you haven't run the tests ("make test").  It would just recurse and crash when it exceeds the max stack depth.
Try doing "man usleep" or "man -k sleep" in cygwin, assuming man pages are installed, to see what sleep functions are available and what you have to include to find them.
If there isn't something like